crypto/rand/drbg_lib.c

```diff
@@ -1101,8 +1101,20 @@ static int drbg_add(const void *buf, int num, double randomness)
 
     buflen = (size_t)num;
 
+#ifdef FIPS_MODE
+    /*
+     * NIST SP-800-90A mandates that entropy *shall not* be provided
+     * by the consuming application, neither for instantiation, nor
+     * for reseeding. By setting the randomness to zero, we ensured
+     * that the buffer contents will be added to the internal state
+     * of the DRBG only as additional data.
+     *
+     * (NIST SP-800-90Ar1, Sections 9.1 and 9.2)
+     */
+    randomness = 0.0;
+#endif
     if (buflen < seedlen || randomness < (double) seedlen) {
-#if defined(OPENSSL_RAND_SEED_NONE)
+# if defined(OPENSSL_RAND_SEED_NONE)
```

Comment:
This shouldn't have the extra indent.
That or the #ifdef above should.

Reply:
I reverted the reindentation.